src/snowflake/cli/plugins/nativeapp/codegen/snowpark/extension_function_utils.py

```diff
@@ -0,0 +1,368 @@
+            )
+
+        if (
+            len(ex_fn[attr].strip()) > 0
+        ):  # To prevent where attr value is "  " etc, which should still be invalid
+            assign_to_none = False
+            if make_uppercase:
+                ex_fn[attr] = ex_fn[attr].upper()
+
+    if assign_to_none:
+        ex_fn[attr] = None
+
+    if assign_to_none and raise_err:
+        raise MalformedExtensionFunctionError(
+            _create_missing_attr_message(attribute=attr, py_file=py_file)
+        )
+
+
+def _sanitize_list_or_dict_attribute(
+    ex_fn: Dict[str, Any],
+    attr: str,
+    expected_type: Type,
+    default_value: Any = None,
+    py_file: Optional[Path] = None,
+    raise_err: bool = False,
+):
+    """
+    Sanitizes a single key-value pair of the specified dictionary. As part of the sanitization,
+    it goes through a few checks. A key must be created if it does not already exist.
+    Then, it checks the type of the value of the key. It also checks for the length of the value, which is why the value must be of type list or dict.
+    A user is able to specity a default value that they want to assign a newly created key to.
+    If any of the sanitization checks fail and the user wants to raise an error, it throws a MalformedExtensionFunctionError.
+    """
+    assign_to_default = True
+    if ex_fn.get(attr, None):
+        if not isinstance(ex_fn[attr], expected_type):
+            raise MalformedExtensionFunctionError(
+                f"Attribute '{attr}' of extension function must be of type '{expected_type}'."
+            )
+
+        if len(ex_fn[attr]) > 0:
+            assign_to_default = False
+
+    if assign_to_default:
+        ex_fn[attr] = default_value
+
+    if assign_to_default and raise_err:
+        raise MalformedExtensionFunctionError(
+            _create_missing_attr_message(attribute=attr, py_file=py_file)
+        )
+
+
+def _create_missing_attr_message(attribute: str, py_file: Optional[Path]):
+    """
+    This message string is used to create an instance of the MalformedExtensionFunctionError.
+    """
+    if py_file is None:
+        raise ValueError("Python file path must not be None.")
+    return f"Required attribute '{attribute}' of extension function is missing or incorrectly defined for an extension function defined in python file {py_file.absolute()}."
+
+
+def _is_function_wellformed(ex_fn: Dict[str, Any]) -> bool:
+    """
+    Checks if the specified dictionary contains a key called 'func'.
+    if it does, then the value must be of type str or a list of fixes size 2.
+    It further checks the item at 1st index of this list.
+    """
+    if ex_fn.get("func", None):
+        if isinstance(ex_fn["func"], str):
+            return ex_fn["func"].strip() != ""
+        elif isinstance(ex_fn["func"], list) and (len(ex_fn["func"]) == 2):
+            return isinstance(ex_fn["func"][1], str) and ex_fn["func"][1].strip() != ""
```

Comment:
Why do we only have to check the 2nd string?

Reply:
There are two types that you could have passed to a func property in Snowpark: Callable and Tuple. 
The tuple construction is this: (file_path, func_name). We don't really do any validation on the file_path yet in our MVP. That leaves the func_name, which we are checking to ensure that is not blank.

Reply:
We actually never use the func when it is of type `Tuple`. There is an exception thrown in one of the other methods `_get_handler()` if it detects the Tuple. What we can do is bring that exception to this method instead of further down. But as it stands, we don't need to perform file validation because we don't act on this func type.